packages/eas-cli/src/project/discourageExpoGoForProd.ts

```diff
@@ -0,0 +1,73 @@
+import { Workflow } from '@expo/eas-build-job';
+import chalk from 'chalk';
+import getenv from 'getenv';
+
+import { resolveWorkflowPerPlatformAsync } from './workflow';
+import { isExpoDevClientInstalled } from '../build/utils/devClient';
+import Log, { learnMore } from '../log';
+import type { ProfileData } from '../utils/profiles';
+import type { Client } from '../vcs/vcs';
+
+const suppressionEnvVarName = 'EAS_BUILD_NO_EXPO_GO_WARNING';
+
+export async function discourageExpoGoForProdAsync(
+  buildProfiles: ProfileData<'build'>[] | undefined,
+  projectDir: string,
+  vcsClient: Client
+): Promise<void> {
+  try {
+    const isExpoGoProdBuild = await detectExpoGoProdBuildAsync(
+      buildProfiles,
+      projectDir,
+      vcsClient
+    );
+    if (!isExpoGoProdBuild) {
+      return;
+    }
+    Log.newLine();
+    Log.warn(
+      `⚠️ It appears you're trying to build an app based on Expo Go for production. Expo Go is not a suitable environment for production apps.`
+    );
+    Log.warn(
+      learnMore('https://docs.expo.dev/develop/development-builds/expo-go-to-dev-build/', {
+        learnMoreMessage: 'Learn more about converting from Expo Go to a development build',
+        dim: false,
+      })
+    );
```

Comment:
i'll think about how to improve the wording on this and get back to you, let's hold off on merging for now